src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -292,13 +292,15 @@ private static ArraySegment<byte> ReadToEnd(Stream stream)
             int written = 0;
             byte[] rented = null;
 
+            ReadOnlySpan<byte> utf8Bom = JsonConstants.Utf8Bom;
```

Comment:
Do you think `ReadToEnd` should be public available (maybe a static on stream/extension method)?

It is being used here as well (something we need to consider updating): https://github.com/dotnet/core-setup/blob/45f9401bf62faf0d3446cfd8681d35cc3487367a/src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Utf8JsonReader.cs#L39-L89

Reply:
Maybe?  But if it was public it would not have this change; this is particular to us reading a stream (which is "bytes" and could be from File.OpenRead) and interpreting the data as UTF-8 text without going through the Encoding classes.

A more general fix (if we wanted to support UTF-16(BOM)) would be to ask StreamReader to do encoding sniffing, then translate all of the chars back through Encoding.UTF8.

DependencyModel would need the change if they similarly want to support BOM-prepended UTF-8 files; but if they don't write BOMs they wouldn't need it.

Reply:
> DependencyModel would need the change if they similarly want to support BOM-prepended UTF-8 files; but if they don't write BOMs they wouldn't need it.

 @eerhardt, does DependencyModel need to support files with the BOM present when reading?

Reply:
It just used `JsonTextReader` from Newtonsoft previously. Does that handle BOMs?

Reply:
> It just used JsonTextReader from Newtonsoft previously. Does that handle BOMs?

Technically, no.  But the StreamReader adapter between the Stream and JsonTextReader does :smile:.  So this change _should_ be ported.